### ISSUE TITLE
tf.nn.relu on nan inputs returns zeros on GPU

### ISSUE THREAD

**Issue by msoelch** [tensorflow/tensorflow#40072]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes, see below.
- OS Platform and Distribution: Ubuntu 16.04, Google Colab
- TensorFlow installed from: binary
- TensorFlow version: 2.2
- Python version: 3.6.9
- CUDA/cuDNN version: 10.1, 7.6.5
- GPU model and memory: GeForce GTX 1080 (8GB) (and Google Colab)

**Describe the current behavior**

The behavior of `tf.nn.relu` when fed with `nan`-valued inputs is inconsistent between GPU and CPU:

- The CPU returns `nan` (as expected).
- The GPU returns `0`, obfuscating `nan` in the computation.

Similar (but not quite identical) behavior has been described for `TF1.14` in #32730 and subsequently fixed in `TF1.15.3`, but remains in `TF2.2`

**Describe the expected behavior**

All devices should show consistent behavior like the CPU, returning `nan`.

**Standalone code to reproduce the issue**
The following snippet can also be found in [this Colab notebook](https://colab.research.google.com/drive/1LtUP4TySBPd7Ng80qdQtB2UIl3mtF3cX?usp=sharing).

The first three assertions do not fail. The last three assertions all fail. This is a bit different from #32730, where the `x1` assertion worked.
```python
x1 = tf.nn.relu(np.nan)
x2 = tf.nn.relu(np.nan * tf.random.normal(shape=[]))
x3 = tf.nn.relu(tf.Variable(np.nan))
with tf.device("/cpu:0"):
    x1_cpu = tf.nn.relu(np.nan)
    x2_cpu = tf.nn.relu(np.nan * tf.random.normal(shape=[]))
    x3_cpu = tf.nn.relu(tf.Variable(np.nan))

assert np.all(np.isnan(x1_cpu.numpy()))
assert np.all(np.isnan(x2_cpu.numpy()))
assert np.all(np.isnan(x3_cpu.numpy()))

assert np.all(np.isnan(x1.numpy()))
assert np.all(np.isnan(x2.numpy()))
assert np.all(np.isnan(x3.numpy()))
```

**Comment by Saduf2019** [comment]
I am able to replicate the issue, please find the [gist here](https://colab.sandbox.google.com/gist/Saduf2019/8c1d1b6eef1745a6d5e6d992f27e65db/untitled208.ipynb) Thanks!

**Comment by bhack** [comment]
Please be aware of https://github.com/tensorflow/tensorflow/issues/42885#issuecomment-687151509 if you are going to retest this on colab again. 
That duplicated issue was closed as solved but it isn't solved.

**Comment by jvishnuvardhan** [comment]
@msoelch Looks like this was resolved in `2.4rc3` that was just released. Can you please check and let us know whether it was resolved or not for you. [Here](https://colab.research.google.com/gist/jvishnuvardhan/e9db6d904cd5c816f78ec7c53bd4962c/untitled208.ipynb) is the gist for our reference. Thanks!

**Comment by msoelch** [comment]
Looks good, thank you!
